MdeModulePkg/Include/Guid/DxeMemoryProtectionSettings.h

```diff
@@ -456,7 +456,8 @@ extern GUID  gDxeMemoryProtectionSettingsGuid;
 #define DXE_MEMORY_PROTECTION_SETTINGS_OFF                      \
           {                                                     \
             DXE_MEMORY_PROTECTION_SETTINGS_CURRENT_VERSION,     \
-            FALSE,   /* Stack Guard On */                       \
+            FALSE,   /* Stack Guard Off */                      \
+            FALSE,   /* Free Memory Guard Off */                \

```

Comment:
I'm honestly tempted to not make it a choice. This will blow platforms up to be sure. But this is also something that they really need to fix. I understand this is a major change, so I'm ok with having this configurable for now, but I feel like in the next iteration or two we should drop the configurability, unless we find a hard block in a bootloader or something.

Reply:
Agreed on all points. I think this may cause issues in the short term for platforms which are misusing free memory so it's good to have a policy configuration but hopefully they will be ironed out soon and we can remove this option.